archiver.go

```diff
@@ -1,33 +1,172 @@
 package debos
 
 import (
-	"log"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
-func TarOptions(compression string) string {
+type ArchiveType int
+
+// Supported types
+const (
+	_ ArchiveType = iota // Guess archive type from file extension
+	Tar
+	Zip
+	Deb
+)
+
+type Archive struct {
+	File        string // Path to archive file
+	Type        ArchiveType
+	compression string
+}
+
+type Archiver interface {
+	CommandUnpack(destination string, options ...string) []string
```

Comment:
CommandUnpack should not be part of the interface if your interface users aren't meant to call it. Implment it onthe base archive class instead (same as a common unpack).

Reply:
But they are. Test is the user for example.
The main idea was to have own method for command line for each archive type, so  any new type has it's own implementation.
In case if base type will deal with all types of archive -- it have no sense to keep all other types.

Reply:
No what i mean is that there is no point of the *interface* which is exposed to the user of the code to have the method. 

The Tests can do a type assertion to get at the specific implementation of a type if they want to test CommandUnpack.

Note that this doesn't mean you need a commaon CommandUnpack, see for example how the BaseAction has some dummy functions which get overriden in the Specific actions to do the specialisations.